### ISSUE TITLE
@clayui/tabs: How do conditionally render tab items?

### ISSUE THREAD

**Issue by pat270** [liferay/clay#5169]
https://issues.liferay.com/browse/COMMERCE-10270 is using ClayTabs and wants to conditionally render ClayTabs.Items only if panels have content in them.  Trying to conditionally render a tab item in https://clayui.com/docs/components/tabs.html with:
```
{false && (
  <ClayTabs.Item
      innerProps={{
        "aria-controls": "tabpanel-2"
      }}
    >
      Tab 2
    </ClayTabs.Item>
)}
```
throws an error:
```
TypeError: Cannot read properties of null (reading 'props')
```

**Comment by brianikim** [comment]
the ideal option IMO is for the ClayTabs code to filter the react children before running the map function, that way you don't have to write that extra code

**Comment by matuzalemsteles** [comment]
Oh yes this is a bug, we don't need to do an extra filter just to check if the child is a valid element, just check if the child is a valid React element before doing the clone.

**Comment by matuzalemsteles** [comment]
@brianikim Just an extra, you can also handle this in other ways and that would be better because Tabs uses declaration order to link the tab with the panel, so you have to make the same condition for the panel that has this logic otherwise can have some side effects if the tab appears during the application lifecycle.

Another simpler option to not add conditional would be to bring all your tabs into an Array.

```jsx
const tabs = [
  {
    name: Liferay.Language.get('full-description'),
    content: 'Something text ou React component'
  },
  ...
];

const [active, setActive] = useState(0);

return (
  <>
    <ClayTabs active={active} modern onActiveChange={setActive}>
      {tabs.map((tab, index) => (
        <ClayTabs.Item
          innerProps={{
            'aria-controls': `tabpanel-${index}`
          }}
        >
          {tab.name}
        </ClayTabs.Item>
      ))}
    </ClayTabs>
    <ClayTabs.Content activeIndex={active} fade>
      {tabs.map((tab, index) => (
        <ClayTabs.TabPane aria-labelledby={`tabpanel-${index}`}>
          {tab.content}
        </ClayTabs.TabPane>
      ))}
    </ClayTabs.Content>
  </>
);
```

You add the tabs that should be displayed in the array, another option would be to add the tab with the disabled property as well.

**Comment by brianikim** [comment]
@matuzalemsteles , disabled is not the way to go for my particular scenario. We require the tab to not exist. Disabled would still show the tab (correct me if I am wrong here)

For tabs into array, is your recommendation to basically do the condition checks prior to adding the tabs into array? You want the return to only be a render right?

Sincerely,
Brian

**Comment by matuzalemsteles** [comment]
> For tabs into array, is your recommendation to basically do the condition checks prior to adding the tabs into array? You want the return to only be a render right?

Yeah, for example you can add `tabs` inside a `useMemo` which has the rules of which tab should be visible and returns the array, if that state needs to change, you can add some property dependency to `useMemo` that changes during rendering for example, this gets simpler too.

```js
const tabs = useMemo(() => {
  const tabs = [];

  if (conditional) {
    tabs.push({
      name: Liferay.Language.get('full-description'),
      content: 'Something text ou React component'
    });
  }

  return tabs;
}, []);
```

> Disabled would still show the tab, just greyed out (correct me if I am wrong here)

Yes, it will still be visible but the user cannot change the tab, that would be another scenario but if yours is not ideal, no problem.

**Comment by brianikim** [comment]
@matuzalemsteles ,

Thank you for the hints. That looks significantly prettier than having to do the checks individually on the return. I will apply the recommendation.
